src/corefx/System.Globalization.Native/pal_calendarData.c

```diff
@@ -324,23 +281,31 @@ bool InvokeCallbackForDateTimePattern(const char* locale,
 {
     UErrorCode err = U_ZERO_ERROR;
     UDateTimePatternGenerator* pGenerator = udatpg_open(locale, &err);
-    UDateTimePatternGeneratorHolder generatorHolder(pGenerator, err);
 
     if (U_FAILURE(err))
+    {
+        udatpg_close(pGenerator);
         return false;
+    }
 
     UErrorCode ignore = U_ZERO_ERROR;
-    int32_t patternLen = udatpg_getBestPattern(pGenerator, patternSkeleton, -1, nullptr, 0, &ignore);
+    int32_t patternLen = udatpg_getBestPattern(pGenerator, patternSkeleton, -1, NULL, 0, &ignore) + 1;
 
-    std::vector<UChar> bestPattern(patternLen + 1, '\0');
+    UChar* bestPattern = calloc(patternLen, sizeof(UChar*));
```

Comment:
Why are you allocating one less char than the original version?

Reply:
I added `+ 1` in the line above.